packages/router/src/router.tsx

```diff
@@ -144,29 +146,35 @@ const InternalRoute: React.VFC<InternalRouteProps> = ({
   )
 }
 
-interface PrivateProps {
+type PrivateProps<P> = Omit<SetProps<P>, 'wrap'> & {
   /** The page name where a user will be redirected when not authenticated */
   unauthenticated: string
   role?: string | string[]
+  wrap?: WrapperType<P> | WrapperType<P>[]
 }
 
 /**
  * `Routes` nested in `Private` require authentication.
  * When a user is not authenticated and attempts to visit this route they will be
  * redirected to `unauthenticated` route.
  */
-const Private: React.FC<PrivateProps> = ({
-  children,
-  unauthenticated,
-  role,
-}) => {
+function Private<WrapperProps>(props: PrivateProps<WrapperProps>) {
+  const { role, unauthenticated, children, wrap, ...rest } = props
+
   return (
     <PrivateContextProvider
       isPrivate={true}
       role={role}
       unauthenticated={unauthenticated}
     >
-      {children}
+      {wrap ? (
+        // @ts-expect-error - Set
```

Comment:
Without this comment, this is the error I get

```
Type '{ wrap: WrapperType<WrapperProps> | WrapperType<WrapperProps>[]; } & Omit<PrivateProps<WrapperProps>, "wrap" | ... 2 more ... | "unauthenticated"> & { ...; }' is not assignable to type 'IntrinsicAttributes & WrapperProps & { wrap: WrapperType<WrapperProps> | WrapperType<WrapperProps>[]; children: ReactNode; }'.
  Type '{ wrap: WrapperType<WrapperProps> | WrapperType<WrapperProps>[]; } & Omit<PrivateProps<WrapperProps>, "wrap" | ... 2 more ... | "unauthenticated"> & { ...; }' is not assignable to type 'WrapperProps'.
    'WrapperProps' could be instantiated with an arbitrary type which could be unrelated to '{ wrap: WrapperType<WrapperProps> | WrapperType<WrapperProps>[]; } & Omit<PrivateProps<WrapperProps>, "wrap" | ... 2 more ... | "unauthenticated"> & { ...; }'.ts(2322)
```

@dac09 If you want to flex your TS muscles, please help here 💪